blockchain/chain_sync/src/sync.rs

```diff
@@ -461,13 +461,46 @@ where
         }
         Ok(())
     }
-
-    async fn sync_fork(&mut self, _head: &Tipset, _to: &Tipset) -> Result<Vec<Arc<Tipset>>, Error> {
-        // TODO sync fork until tipsets are equal or reaches genesis
-        todo!()
+    /// fork detected, collect tipsets to be included in return_set sync_headers_reverse
+    async fn sync_fork(&mut self, head: &Tipset, to: &Tipset) -> Result<Vec<Tipset>, Error> {
+        // TODO change from using random peerID to managed
+        let peer_id = PeerId::random();
+        // pulled from Lotus: https://github.com/filecoin-project/lotus/blob/master/chain/sync.go#L996
+        const FORK_LENGTH_THRESHOLD: u64 = 500;
+
+        // Load blocks from network using blocksync
+        let tips: Vec<Tipset> = self
+            .network
+            .blocksync_headers(peer_id.clone(), head.parents(), FORK_LENGTH_THRESHOLD)
+            .await
+            .map_err(|_| Error::Other("Could not retrieve tipset".to_string()))?;
+
+        let mut ts = self.chain_store.tipset_from_keys(to.parents())?;
+
+        for i in 0..tips.len() {
+            if *ts.tip_epoch().chain_epoch() == 0 {
```

Comment:
the chain_epoch function name is confusing, since it is going from a chain_epoch to a u64, also the `tip_epoch()` could have just been `epoch()` since it's operating on a tipset (nits aside don't need to change in this PR, just thought I'd give thoughts on API)